source/lac/trilinos_block_sparse_matrix.cc

```diff
@@ -258,7 +258,7 @@ namespace TrilinosWrappers
   void
   BlockSparseMatrix::collect_sizes ()
   {
-    compress();
```

Comment:
I think that in this case I really don't need it because it doesn't actually access individual elements of the matrix. It only accesses the sizes of the individual blocks which are never affected by `compress` on the blocks.